sources/osgAnimation/UpdateMorph.js

```diff
@@ -47,21 +56,174 @@ define( [
             return this._isInitialized;
         },
 
-        getTarget: function ( index ) {
-            return this._targets[ index ];
+        getNumTarget: function () {
+            return this._targets.length;
         },
 
-        getNumTarget: function () {
-            return this._targetName.length;
+        getTarget: function ( index ) {
+            return this._targets[ index ];
         },
 
         getTargetName: function ( index ) {
-            return this._targetName[ index ];
+            return this._targetNames[ index ];
         },
 
         addTarget: function ( name, index ) {
             this._targets[ index ] = Target.createFloatTarget( 0 );
-            this._targetName[ index ] = name;
+            this._targetNames[ index ] = name;
+        },
+
+        _remapBufferArrays: function () {
+
+            // basically, this function remaps all the active morphed VA to the 4th first morphTargets VA
+            var indexMap = this._indexMap;
+            var morphs = this._morphs;
+            for ( var i = 0; i < 4; ++i ) {
+                var index = indexMap[ i ];
+
+                for ( var j = 0, nbMorphs = morphs.length; j < nbMorphs; ++j ) {
+
+                    var morph = morphs[ j ];
+                    var vAttrs = morph.getVertexAttributeList();
+                    var morphNames = morph.getMorphTargetNames();
+                    for ( var k = 0, nbNames = morphNames.length; k < nbNames; ++k ) {
+
+                        var attName = morphNames[ k ];
+                        vAttrs[ attName + '_' + i ].setActiveArray( vAttrs[ attName + '_' + index ] );
```

Comment:
"IF" it's done each frame, caching "'_'+i" and "'_'+index" two loop above can save a bit of gc/string/itoa compute